src/Workspaces/Core/Portable/SolutionCrawler/IIncrementalAnalyzer2.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.SolutionCrawler
+{
+    // CONSIDER: We can merge IIncrementalAnalyzer2 with IIncrementalAnalyzer once all of our
+    //           IVT partners that use IIncrementalAnalyzer have migrated to ExternalAccess layer.
+    internal interface IIncrementalAnalyzer2 : IIncrementalAnalyzer
```

Comment:
Can we do a test insertion to verify that this does not break partners? who currently uses `IIncrementalAnalyzer` via IVTs and have we filed bugs on them? Woud prefer to just update the `IIncrementalAnalyzer` interface

Reply:
AFAIK, only the unit test team used this interface outside of Roslyn. @tmat @CyrusNajmabadi @Cosifne if things have changed now and they have moved to external access and no longer are broken if we change this interface.

Reply:
Seems Unit testing team has already moved their incremental analyzer to ExternalAccess layer: https://github.com/dotnet/roslyn/blob/c9e34615aba35f052fafe8ae783092d366d06c06/src/Workspaces/Core/Portable/ExternalAccess/UnitTesting/UnitTestingIncrementalAnalyzer.cs#L11-L15

Reply:
@jmarolf I filed https://github.com/dotnet/roslyn/issues/45936 to decouple this cleanup change from the core functionality add in this PR. As a follow-up PR, we will update the code to merge `IIncrementalAnalyzer2` and `IIncrementalAnalyzer`, introduce an `AbstractIncrementalAnalyzer` to default non-source document implementations to no-op and also do a test insertion to verify no partner teams are broken.

Reply:
On the Unit Testing side it only depends on the interface in the IVT layer so it won't be a problem to change the original incremental analzyer